File: src/Services/Catalog/Catalog.API/Entities/Product.cs
namespace Catalog.API.Entities
{
    public class Product : BaseEntity
    {
        public long Id { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public Nullable<decimal> Price { get; set; }
        public decimal MRP { get; set; }
        public long CatalogTypeId { get; set; }
        public CatalogType CatalogType { get; set; }
        public long CatalogBrandId { get; set; }
        public CatalogBrand CatalogBrand { get; set; }

        // Quantity in stock
        public int AvailableStock { get; set; }

        // Available stock at which we should reorder
        public int RestockThreshold { get; set; }
        public int MaxStockThreshold { get; set; }
        //public bool OnReorder { get; set; }
        public Image Image { get; set; }

        public Product() { }
    }
}

File: src/UI/ShopperHub/src/models/product.ts
export class Product {
    constructor(
        id: number,
        name: string,
        description: string,
        price: number,
        mrp: number,
        catalogTypeId: number,
        catalogType: string,
        catalogBrandId: number,
        catalogBrand: string,
        availableStock: number,
        restockThreshold: number,
        maxStockThreshold: number,
        imageId: number,
        imageName: string,
        imageCaption: string,
        imageData: string
    ) {
        this.id = id;
        this.name = name;
        this.description = description;
        this.price = price;
        this.mrp = mrp;

        this.imageCaption = imageCaption;
        this.imageName = imageName;
        this.imageData = imageData;
    }

    id: number;
    name: string;
    description: string;
    price: number;
    mrp: number;
    catalogTypeId: number;
    catalogType: string;
    catalogBrandId: number;
    catalogBrand: string;
    availableStock: number;
    restockThreshold: number;
    maxStockThreshold: number;
    imageId: number;
    imageName: string;
    imageCaption: string;
    imageData: string;
}

export type ProductType = {

    id: number;
    name: string;
    description: string;
    price: number;
    mrp: number;
    imageName: string;
    imageCaption: string;
    imageData: string;
    availableStock: number;
    count: number;
}